### DISCUSSION TITLE
[2024-04-02] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by Frizi** [enso-org/enso#9603] [Weekly Demo Day]
- Removed significant portion of remaining old GUI Rust modules <img width="109" alt="image" src="https://github.com/enso-org/enso/assets/919491/a6913f8f-d15b-457e-9aeb-caa2aa2c33b5"> #9540
- Continued attempts to fix reconnection issues after hibernation, but decided to switch away to other issues and leave that for others.
- Added proper application teardown after dashboard closes the project, closing all existing network connections and stopping attempted reconnections #8964
- Started working on recovering client-side state after refresh #6250
- Debugged new issues with reconnecting with engine after page refresh caused by recent Dashboard changes, partially fixed in #9604
- Fixed smaller bugs, such as:
  - adjusted minimum visualization size #9515
  - Prevent handling repeated events when holding a keyboard button #8538
  - Fixed vector editor appearing on non-vector expressions #9596

**Comment by vitvakatu** [comment]
@farmaazon 

- https://github.com/enso-org/enso/pull/9476

[Screencast from 2024-03-22 09-09-07.webm](https://github.com/enso-org/enso/assets/3919101/6ad86145-6882-4bde-993d-b1270f1ec06c)

- https://github.com/enso-org/enso/pull/9495
- https://github.com/enso-org/enso/pull/9498
- https://github.com/enso-org/enso/pull/9536

[Screencast from 2024-03-25 15-44-50.webm](https://github.com/enso-org/enso/assets/3919101/bd5b2bbd-631f-4397-97ae-d925b9194210)

- https://github.com/enso-org/enso/pull/9538
- https://github.com/enso-org/enso/pull/9545
- https://github.com/enso-org/enso/pull/9561
- https://github.com/enso-org/enso/pull/9569

---

@vitvakatu 

- https://github.com/enso-org/enso/pull/9570

https://github.com/enso-org/enso/assets/6566674/a4497bea-f432-486d-85f8-f2772ba5694f

- https://github.com/enso-org/enso/pull/9578
- https://github.com/enso-org/enso/pull/9594
- https://github.com/enso-org/enso/pull/9585

https://github.com/enso-org/enso/assets/6566674/3418d31e-2d80-48b2-ba8b-9e000b09cf14

- https://github.com/enso-org/enso/pull/9602

**Comment by kazcw** [comment]
- Multi-select widget (#9561):

https://github.com/enso-org/enso/assets/1047859/49a2ce59-d81a-4bf3-b314-4771b8adf925

- Interaction resumption (PR today):

https://github.com/enso-org/enso/assets/1047859/bcc7fa02-847f-4cd0-b9fc-3186a97c5816

- Fix a bug caught by prop-tests: #9513
- Add hoverable tracks to scrollbars: #9461

**Comment by mwu-tow** [comment]
CI improvements:
* Backend:
    * https://github.com/enso-org/enso/pull/9309
    * https://github.com/enso-org/enso/pull/9506
    * https://github.com/enso-org/enso/pull/9580
* GUI:
    * https://github.com/enso-org/enso/pull/9507
    * https://github.com/enso-org/enso/pull/9557
    * https://github.com/enso-org/enso/pull/9544

* Other:
    * https://github.com/enso-org/enso/pull/9517
    * https://github.com/enso-org/enso/pull/9564
    * https://github.com/enso-org/enso/pull/9595

Current matters:
* No issues with nightlies.
* Stability overall improved, still there are transient failures on macOS runners.
* Ongoing discussion about CI future.

**Comment by JaroslavTulach** [comment]
Engine team achievements:
- work in progress on #9528
- _still pending, but progressing_ - #9452
- `import` & _FQNs_ **done**: #9539
- Subprocess does not freeze: #9550
- [Towards Incremental Compilation](https://docs.google.com/document/d/1bKX2swSW-6ACLkTt623adrT45-SeANom4vv3w6XvqKc) _while avoiding UUID metadata meanwhile…_ vision ready for your review
